packages/react-component-library/src/components/DatePicker/DatePickerInput.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react'
+import { isString } from 'lodash'
+import { addHours, isMatch, parse } from 'date-fns'
+import { DayPickerProps, ModifiersUtils } from 'react-day-picker'
+
+import { ComponentWithClass } from '../../common/ComponentWithClass'
+import { StyledInput } from './partials/StyledInput'
+import { useInputValue } from './useInputValue'
+
+const VALUE_PATTERN = /^[0-9]{1,2}\/[0-9]{1,2}\/[0-9]{4}$/
```

Comment:
Could we call this `INPUT_MASK` maybe instead of `VALUE_PATTERN`?

Reply:
- **Update picker as soon as full valid date is entered:** That would be ideal, if not too much dev effort. I think it would also partially address the next point.

- **Hitting return closes the picker:** The reason behind this behaviour is that, when using a keyboard and after having typed a date, the user shouldn't have to tab through the picker elements in order to move to the next field. Am I right that if we can implement the first point, we will address your concern here, as the picker will update automatically, before even pressing Return?